Public/Src/Engine/Processes/AugmentedManifestReporter.cs

```diff
@@ -0,0 +1,204 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.ContractsLight;
+using System.Globalization;
+using System.Text;
+using BuildXL.Native.IO;
+using JetBrains.Annotations;
+using Microsoft.Win32.SafeHandles;
+
+namespace BuildXL.Processes
+{
+    /// <summary>
+    /// This class is the entry point for trusted tools to report file accesses directly to the sandbox, without
+    /// actually performing any IO
+    /// </summary>
+    /// <remarks>
+    /// The main use case for this reporter is when pips are configured to allow trusted processes to breakaway from the sandbox. In that circumstance,
+    /// processes may need to compensate for some unobserved accesses.
+    /// In order for processes to use this reporter, they must run in the context of a pip which has a non-empty set of processes that 
+    /// are allowed to breakaway. Otherwise, attempting to report an access will fail.
+    /// Observe this reporter is only intended for internal (to BuildXL) tools at this point, since it requires tools to link against
+    /// this particular BuildXL library. In the future we may consider a more general approach for arbitrary (trusted) tools to perform these actions
+    /// </remarks>
+    public sealed class AugmentedManifestReporter
+    {
+        [CanBeNull]
+        private readonly SafeFileHandle m_detoursReportHandle = null;

```

Comment:
nit: It might be a little cleaner exclusively initialize readonly variables in the constructor (and there assign `null` when appropriate)

Reply:
Will do

---
In reply to: [343769663](https://github.com/microsoft/BuildXL/pull/1175#discussion_r343769663) [](ancestors = 343769663)